internal/infrastructure/elastic_search/elastic_search_service.go

```diff
@@ -0,0 +1,113 @@
+package elasticsearch
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/DIMO-Network/device-definitions-api/internal/config"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog"
+)
+
+type ElasticSearch struct {
+	BaseURL        string
+	Token          string
+	MetaEngineName string
+	httpClient     *http.Client
+	log            zerolog.Logger
+}
+
+func NewElasticSearch(settings *config.Settings, logger zerolog.Logger) (*ElasticSearch, error) {
+	var netClient = &http.Client{
+		Timeout: time.Second * 10,
+	}
+	return &ElasticSearch{
+		BaseURL:        settings.ElasticSearchAppSearchHost,
+		Token:          settings.ElasticSearchAppSearchToken,
```

Comment:
We'll want to create settings with more appropriate names (we're not using "app search").